tests/basic/filter_test.py

```diff
@@ -86,6 +86,46 @@ def test_combined_filter_with_monitor_type():
         assert ensure_always(lambda: not has_datapoint_with_metric_name(backend, "uptime"), 5)
 
 
+# Test overlapping filters with different negation. Blacklist is favored
+def test_overlapping_filter_with_monitor_type():
+    with run_agent(
+        """
+monitors:
+  - type: collectd/memory
+  - type: collectd/uptime
+metricsToExclude:
+  - metricName: uptime
+    negated: true
+    monitorType: collectd/uptime
+  - metricName: uptime
+    monitorType: collectd/uptime
+"""
+    ) as [backend, _, _]:
+        assert wait_for(lambda: has_datapoint_with_metric_name(backend, "memory.used"))
+        assert wait_for(lambda: has_datapoint_with_metric_name(backend, "memory.free"))
+        assert ensure_always(lambda: not has_datapoint_with_metric_name(backend, "uptime"), 5)
+
+
+# Test overlapping filters with different negation. Blacklist is favored
+def test_overlapping_filter_with_monitor_type2():
+    with run_agent(
+        """
+monitors:
+  - type: collectd/memory
+  - type: collectd/uptime
+metricsToExclude:
+  - metricName: uptime
+    monitorType: collectd/uptime
+  - metricName: uptime
```

Comment:
What does this mean for two contradictory filters?

Reply:
test_overlapping_filter_with_monitor_type and test_overlapping_filter_with_monitor_type2
show that order in which filters are defined does not matter, and that exclusion of metrics will always take precedence over "negated:true"